app/Http/Controllers/TransportController.php

```diff
@@ -633,17 +633,22 @@ public static function usageByDay(Carbon $date) {
      * @param Carbon $start
      * @param Carbon $end
      * @return Collection
+     * @see https://stackoverflow.com/questions/53697172/laravel-eloquent-query-to-check-overlapping-start-and-end-datetime-fields/53697498
      */
     private static function getOverlappingCheckIns(User $user, Carbon $start, Carbon $end): Collection {
-        $start = $start->clone()->subMinutes(2);
-        $end   = $end->clone()->addMinutes(2);
 
         $user->loadMissing(['statuses', 'statuses.trainCheckin']);
 
+        //increase the tolerance for start and end of collisions
+        $start = $start->addMinutes(2);
+        $end   = $end->subMinutes(2);
+
         return $user->statuses->map(function($status) {
             return $status->trainCheckin;
         })->filter(function($trainCheckIn) use ($start, $end) {
-            return $trainCheckIn->arrival->isBetween($start, $end) || $trainCheckIn->departure->isBetween($start, $end);
+            return ($trainCheckIn->arrival > $start && $trainCheckIn->departure < $end)
+                || ($trainCheckIn->arrival > $end && $trainCheckIn->departure < $start)
+                || ($trainCheckIn->departure > $start && $trainCheckIn->arrival < $start);
         });
```

Comment:
Ist das ein Rollback von Kris's Änderungen?

Reply:
Yes and no. I forgot a case in my code, Levin has added it now with these. (See Case 3 in his test cases)